### ISSUE TITLE
Allow multiple interpreters for a pex

### ISSUE THREAD

**Issue by a-recknagel** [pantsbuild/pex#1477]
Hi,

I think my issue is with documentation, because from reading https://github.com/pantsbuild/pex/issues/654 I feel like my use case seems to be supported, but I couldn't quickly work out how.

---

So, I have an application with a script-entrypoint that I want to bundle with its dependencies to offer my clients as a single binary that I know will "just work" on their machine without needing to download anything that could potentially break / is untested.

I have build my application and all its dependencies as wheels, put them in a folder called `wheelhouse`, and use this command to build the pex file:

```bash
pex --interpreter-constraint ">=3.8"  -r requirements.txt setuptools my_app -c my_app_entrypoint --no-pypi --repo wheelhouse/ -o my_app.pex
```
I know that it uses a python version 3.8 for the pex command, because it's the only interpreter on the build system, but the application should run just fine under python3.9 as well (which is what I wanted to express with >=3.8). It works fine with python 3.8, but when I run it with only a python3.9 on my PATH, it fails. How can I correctly tell pex during the build that it's ok to use 3.8 or higher?

**Comment by jsirois** [comment]
@a-recknagel it appears your PEX includes at least one platform-specific wheel. You can confirm this with `pex-tools <your PEX> info -i2`. You should see in `distributions` at least one with platform specific tags (like cp38m). If you want to build a PEX that includes the distributions needed by interpreters you don't have on hand, you need to specify a `--platform` for each and every one of those. To see how to write a `--platform`, consult `pex --help` which will include an example for your current platform. Alternatively, use `pex --platform help`.

**Comment by a-recknagel** [comment]
Yeah, that's probably it, I didn't really consider that. Can I create a multi- platform pex by including e.g. cp38 and cp39 wheels of such dependencies in a single one, or should I create multiple pexes instead?

**Comment by jsirois** [comment]
You can create one PEX with all the wheels. If you specify multiple platforms (`--platform X --platform Y ...`), Pex does the right thing. You can again inspect this with `pex-tools <your PEX> info -i2` after building the PEX.

**Comment by a-recknagel** [comment]
Sweet, thanks again. I'll post the right command as soon as I worked it out, in case anyone stumbles over this post via google.
